internal/webhooks/machine_test.go

```diff
@@ -225,3 +229,119 @@ func TestMachineVersionValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestMachineTaintValidation(t *testing.T) {
+	m := builder.Machine("default", "machine1").
+		WithBootstrapTemplate(builder.BootstrapTemplate("default", "bootstrap-template").Build())
+	webhook := &Machine{}
+
+	tests := []struct {
+		name           string
+		machine        *clusterv1.Machine
+		featureEnabled bool
+		expectErr      bool
+	}{
+		{
```

Comment:
Let's add the following cases
* taints specified but feature gate off
* ValidateLabelName => err
* IsValidLabelValue => err
* machine nil for validateMachineTaintsForWorkers